daliuge-engine/dlg/drop.py

```diff
@@ -383,47 +384,69 @@ def __init__(self, oid, uid, **kwargs):
         )  # no need to use synchronised self.status here
 
     def _extract_attributes(self, **kwargs):
+        """
+        Extracts component and app params then assigns them to class instance attributes.
+        Component params take pro
+        """
         def getmembers(object, predicate=None):
             for cls in object.__class__.__mro__[:-1]:
                 for k, v in vars(cls).items():
                     if not predicate or predicate(v):
                         yield k, v
 
+        def get_param_value(attr_name, default_value):
+            has_component_param = attr_name in kwargs
+            has_app_param = hasattr(self, 'parameters') \
+                and 'applicationArgs' in self.parameters \
+                and attr_name in self.parameters['applicationArgs']
+
+            if has_component_param and has_app_param:
+                logger.warning(f"Drop has both component and app param {attr_name}. Using component param.")
+            if has_component_param:
+                param = kwargs.get(attr_name)
+            elif has_app_param:
+                param = self.parameters['applicationArgs'].get(attr_name).value
+            else:
+                param = default_value
+            return param
```

Comment:
This block controls the priority between component params and app params.
There is some complication where I can't find a reason to load `applicationArgs['name'].default_value` since I'd expect eagle to populate an appropriate `applicationArgs['name'].value` else the app default is used.